--- origsrc/openslp-1.2.1/common/Makefile.am	2005-02-07 23:33:21.000000000 -0600
+++ src/openslp-1.2.1/common/Makefile.am	2007-10-29 12:25:47.911154400 -0500
@@ -21,6 +21,7 @@
 	slp_xmalloc.c \
 	slp_xcast.c \
 	slp_iface.c \
+	slp_net.c \
 	slp_parse.c \
 	slp_pid.c \
 	slp_dhcp.c \
--- origsrc/openslp-1.2.1/common/slp_dhcp.c	2005-03-02 11:00:31.000000000 -0600
+++ src/openslp-1.2.1/common/slp_dhcp.c	2007-10-29 11:57:21.859656600 -0500
@@ -65,7 +65,9 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
+#ifndef __CYGWIN__
 #include <net/if_arp.h>
+#endif
 #include <sys/time.h>
 #endif
 
--- origsrc/openslp-1.2.1/libslp/Makefile.am	2005-02-25 18:24:22.000000000 -0600
+++ src/openslp-1.2.1/libslp/Makefile.am	2007-10-29 12:14:25.288673300 -0500
@@ -29,4 +29,4 @@
 #libslp_la_LIBADD = -L../common/libcommonlibslp
 libslp_la_LIBADD = ../common/libcommonlibslp.la
 
-libslp_la_LDFLAGS = -version-info 1:1:0
+libslp_la_LDFLAGS = -version-info 1:1:0 -no-undefined
--- origsrc/openslp-1.2.1/slpd/slpd_main.c	2005-02-08 17:24:53.000000000 -0600
+++ src/openslp-1.2.1/slpd/slpd_main.c	2007-10-29 12:38:17.245383200 -0500
@@ -484,6 +484,7 @@
         SLPDFatal("Invalid command line\n");
     }
 
+#ifndef __CYGWIN__
     /*------------------------------*/
     /* Make sure we are root        */
     /*------------------------------*/
@@ -491,6 +492,7 @@
     {
         SLPDFatal("slpd must be started by root\n");
     }
+#endif
 
     /*--------------------------------------*/
     /* Make sure we are not already running */
